src/containers/SideNav/SideNav.js

```diff
@@ -73,8 +73,20 @@ export class SideNav extends Component {
 
     if (namespace === ALL_NAMESPACES) {
       this.props.selectNamespace(namespace);
-      history.push('/');
-      return;
+      const currentURL = this.props.match.url;
```

Comment:
Ideally this code wouldn't be aware of the URL structure, instead it should use the [router utils](https://github.com/tektoncd/dashboard/blob/master/packages/utils/src/utils/router.js)

- instead of checking if the URL includes '/namespaces/' we should check for `match.params.namespace`
- instead of checking if the URL includes '/taskruns' we should check for `urls.taskruns.all()`
- etc.

This allows us to change the URL structure in a central location without things breaking (apart from the odd test that still needs to be updated to use the utils).

We should also ensure any other affected pages are updated to ensure consistency across the dashboard experience. Is there any way we can do this without needing to explicitly add handling for each affected page?

Reply:
The `namespaces` and `taskruns` was jto ensure it was on the taskruns page as the url will always have those components in to avoid it being confused with another page 

The only pages I noticed the specific issue on was where the url changed to include the `namespace/<namespace>/`. This does appear to happen for PipelineResources and TaskRuns, not sure how we could do this for every page without having to explicity handle each page 

The filters get removed when you change namespace in the dashboard currently. Would be good to keep the filters on when you switch namespace if we assume that a user may have the same task but in multiple namespaces